query/src/main/scala/filodb/query/exec/ExecPlan.scala

```diff
@@ -208,9 +209,23 @@ trait ExecPlan extends QueryCommand {
 
               numResultSamples += srv.numRowsSerialized
               // fail the query instead of limiting range vectors and returning incomplete/inaccurate results
-              if (enforceLimit && numResultSamples > queryContext.plannerParams.sampleLimit)
+              if (enforceSampleLimit && numResultSamples > queryContext.plannerParams.sampleLimit)
                 throw new BadQueryException(s"This query results in more than ${queryContext.plannerParams.
                   sampleLimit} samples. Try applying more filters or reduce time range.")
+
+              resultSize += srv.numBytes() + srv.key.keySize
+              if (resultSize > querySession.qContext.plannerParams.resultByteLimit) {
+                qLogger.warn(s"Reached maximum result size (final or intermediate) for data serialized out of a host " +
+                  s"(${queryContext.plannerParams.resultByteLimit / 1e6} MB). " +
```

Comment:
This may be a hard to read decimal in error message. Could be formatted for better readability.

Also "for data serialized out of a host" could be "for data serialized out of a host or shard"

Reply:
updated